src/AutoDiff.m

```diff
@@ -1088,13 +1088,9 @@
 
 
         function y = det(x)
-            if all(size(x) == [2, 2])
-                y.values = det(x.values);
-                y.derivatives = [x.values(2, 2), -x.values(1, 2), -x.values(2, 1), x.values(1, 1)] * x.derivatives;
-                y = AutoDiff(y.values, y.derivatives);
-            else
-                error('not yet coded')
-            end
+            y.values = det(x.values);
+            y.derivatives = det(x.values).*inv(x.values)';
```

Comment:
There is a part missing in your equation as you are not using `x.derivatives`.
It should be 
`y.derivatives = reshape(det(x.values).*inv(x.values)',1,[]) * x.derivatives;`
Unfortunately my test with `CheckAutoDiffJacobian ` uses `x.derivatives` as the identity so it does not detect this kind of error. I would need to improve the function CheckAutoDiffJacobian to also test a random directional derivatives.

Reply:
Thanks for your reply.

There are three ways to determine this directional derivative in Matlab:
```
y.derivatives = reshape(det(x.values).*inv(x.values)',1,[]) * x.derivatives;
y.derivatives = reshape(det(x.values).*(x.values\eye(size(x.values)))',1,[]) * x.derivatives;
y.derivatives = reshape(adjoint(x.values'),1,[]) * x.derivatives;
```
However, the speed test shows that the third variant is much slower.
```
x = rand(500);
f = det(x);

tic
dfdx1 = det(x).*inv(x)';
toc

tic
dfdx2 = det(x).*(x\eye(size(x)))';
toc

tic
dfdx3 = adjoint(x');
toc
```
```
Elapsed time is 0.005788 seconds.
Elapsed time is 0.006415 seconds.
Elapsed time is 2.654704 seconds.
```